chapters/01_introduction/main.tex

```diff
@@ -1 +1,600 @@
+for common errors and to ensure that the code follows a set of coding standards.
+Table~\ref{tab:code_quality_tools} shows a list of tools that were used
+throughout the development of software and writing of this thesis.
+
+\begin{table}[H]
+    \centering
+    \begin{tabular}{|l|l|}
+        \hline
+        \textbf{Tool} & \textbf{Description} \\
+        \hline
+        \texttt{black} & A code formatter that is used to ensure that the code
+        is \\ & compliant with the PEP 8 coding standard. \\
+        \hline
+        \texttt{flake8} & A tool for style guide enforcement. \\
+        \hline
+        \texttt{pylint} & A code linter that statically analyses your code. \\
+        \hline
+        \texttt{mypy} & A Python static type checker. \\
+        \hline
+        \hline
+        \texttt{aspell} & A grammar checker that catches spelling errors. \\
+        \hline
+        \texttt{alex} & A checker for insensitive and inconsiderate writing. \\
+        \hline
+        \texttt{proselint} & A linter for prose. \\
+        \hline
+    \end{tabular}
+    \caption{List of tools used to check the code quality.}
+    \label{tab:code_quality_tools}
+\end{table}
+
+The tools listed in Table~\ref{tab:code_quality_tools} were able to be
+automated using the \texttt{tox} library.
+Tox is an automation tool that aims to automate and standardise testing in
+Python.
+It is a generic virtual environment management and test command line tool that
+can be used to check if a package installs correctly and run tests.
+It also acts as a frontend to Continuous Integration (CI) servers such as
+\texttt{GitHub Actions}.
+GitHub Actions is a service hosted on GitHub that is used to automate software
+development workflows.
+It is used to build, test and deploy software.
+GitHub Actions was used to automate the testing and code quality checks of the
+software associated with this thesis.
+Figure~\ref{fig:github_actions} shows an example of a GitHub Actions workflow
+that was used to automate the testing and code quality checks of the
+\texttt{ambulance\_game} library.
+
+\begin{figure}[H]
+    \includegraphics[width=\linewidth]{chapters/01_introduction/Bin/github_actions.JPG}
+    \caption{GitHub Actions workflow example.}
+    \label{fig:github_actions}
+\end{figure}
+
+
+\section{Chapter summary}
+
+This chapter has given an introduction to OR and its applications in the
+healthcare sector.
+Additionally, a brief overview of the history of OR has been given, along with
+a discussion on queuing theory and game theory.
+Additionally, the motivation for the research problem has also been discussed
+and the thesis objectives have been defined.
+
+The chapter has also provided an overview of the best practices that were
+adopted throughout the development of the software associated with this thesis.
+Tools such as tox and GitHub Actions were used to automate the testing
+and code quality checks of the software.
+
+
```

Comment:
This is a really nice chapter. I'll be used it as a reference for future students.

Reply:
Thank you 😄